### PR TITLE
Support image forcePull policy for runtime image when do extended build

### PR DESCRIPTION
**Pull request by wanghaoran1988**
To address this [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1363944) , @PI-Victor @php-coder 
, could you please help review ?

### CODE REVIEW

**Comment by mfojtik** [comment]
@bparees I wonder if you can have different policy for runtime and build image, but I guess not (it will require adding new fields to API and make this more complete). So this LGTM.

File: pkg/build/builder/sti.go [hunk 1]
```diff
@@ -227,6 +227,13 @@ func (s *S2IBuilder) Build() error {
 		config.RuntimeImage = runtimeImageName
 		config.RuntimeAuthentication, _ = dockercfg.NewHelper().GetDockerAuth(runtimeImageName, dockercfg.PullAuthType)
 		config.RuntimeArtifacts = copyToVolumeList(s.build.Spec.Strategy.SourceStrategy.RuntimeArtifacts)
+		if s.build.Spec.Strategy.SourceStrategy.ForcePull {
+			glog.V(4).Infof("With force pull true, setting runtime image pull policies to %s", s2iapi.PullAlways)
+			config.RuntimeImagePullPolicy = s2iapi.PullAlways
+		} else {
+			glog.V(4).Infof("With force pull false, setting runtime image pull policies to %s", s2iapi.PullIfNotPresent)
+			config.RuntimeImagePullPolicy = s2iapi.PullIfNotPresent
+		}
```
Comment (php-coder) [inline pkg/build/builder/sti.go]: I would rather log the message once:
```go
glog.V(4).Infof("With force pull %v, setting runtime image pull policy to %s", s.build.Spec.Strategy.SourceStrategy.ForcePull, s2iapi.PullIfNotPresent)
```

Also s/policies/policy/ in the message.

BTW, do we really need this message at all?

**Comment by php-coder** [comment]
LGTM.

**Comment by PI-Victor** [comment]
@wanghaoran1988 nice job. this is my fault, no tests, no bugs discovered :)
@php-coder does this address the issue you were talking about last nite with @bparees ?

File: pkg/build/builder/sti.go (hunk 1 above)
Comment (wanghaoran1988) [inline pkg/build/builder/sti.go]: @php-coder s.build.Spec.Strategy.SourceStrategy.ForcePull sometime will be nil , if we don't set this in the bc, so I think twice of the log is acceptable.

**Comment by mfojtik** [comment]
[merge]

**Comment by php-coder** [comment]
@wanghaoran1988 BTW, would it be better to move this to the same `if` for builder image? https://github.com/openshift/origin/pull/10205/files#diff-b9627ec6ab0689e50e5fa3e5d958c1d3R193

**Comment by bparees** [comment]
> @bparees I wonder if you can have different policy for runtime and build image, but I guess not (it will require adding new fields to API and make this more complete).

@mfojtik yeah not for now.  one of many things we punted on for the first iteration.

**Comment by bparees** [comment]
> @php-coder does this address the issue you were talking about last nite with @bparees ?

@PI-Victor no, that was(is?) an issue with us not using the right auth to pull the image, not setting the pull policy.  @php-coder what did you find out w/ respect to that?

**Comment by php-coder** [comment]
>@PI-Victor no, that was(is?) an issue with us not using the right auth to pull the image, not setting the pull policy.

Yes, it is another issue.

>@php-coder what did you find out w/ respect to that?

In case of using `runtimeImage` env variable `PULL_DOCKERCFG_PATH` is absent. Looks like `build.Spec.Strategy.SourceStrategy.PullSecret` is set to `nil` in this case. I'm searching why and who should set it.

**Comment by wanghaoran1988** [comment]
@php-coder updated

**Comment by php-coder** [comment]
@wanghaoran1988 Thank you! Now it's even better and cleaner.

**Comment by PI-Victor** [comment]
@bparees @php-coder i know why, the secret it never gets mounted in the runtime image, can we move this discussion to a new issue?

**Comment by bparees** [comment]
@PI-Victor yes, please open one.

**Comment by PI-Victor** [comment]
@bparees not really sure why this is stuck, can we merge this?
don't mean to push but i think this is also a p1

**Comment by bparees** [comment]
@PI-Victor it's in the merge queue already, at position 1.

**Comment by PI-Victor** [comment]
@bparees my bad, got confused by:  `Origin Action Required: Please contact #openshift-dev to have this pull request manually reviewed and tested`

**Comment by bparees** [comment]
@PI-Victor you pushed a change after the merge tag was applied, so that took it out of the queue.

**Comment by bparees** [comment]
[merge]

**Pull request merged**